### ISSUE TITLE
Redundancy between ``deepinv.utils.plotting.rescale_img`` and ``deepinv.utils.plotting.preprocess_img``

### ISSUE THREAD

**Issue by Tmodrzyk** [deepinv/deepinv#619]
Both functions have very similar roles and are used interchangeably in the plot functions.
I think it would be better to merge them. For reference, here's the code of both functions:

```python
def preprocess_img(im, rescale_mode="min_max"):
    with torch.no_grad():
        if im.shape[1] == 2:  # for complex images
            pimg = im.pow(2).sum(dim=1, keepdim=True).sqrt().type(torch.float32)
        elif im.shape[1] > 3:
            pimg = im.type(torch.float32)
        else:
            if torch.is_complex(im):
                pimg = im.abs().type(torch.float32)
            else:
                pimg = im.type(torch.float32)

        pimg = rescale_img(pimg, rescale_mode=rescale_mode)
    return pimg 
```
---
```
def rescale_img(im, rescale_mode="min_max"):
    img = im.clone()
    if rescale_mode == "min_max":
        shape = img.shape
        img = img.reshape(shape[0], -1)
        mini = img.min(1)[0]
        maxi = img.max(1)[0]
        idx = mini < maxi
        mini = mini[idx].unsqueeze(1)
        maxi = maxi[idx].unsqueeze(1)
        img[idx, :] = (img[idx, :] - mini) / (maxi - mini)
        img = img.reshape(shape)
    elif rescale_mode == "clip":
        img = img.clamp(min=0.0, max=1.0)
    else:
        raise ValueError("rescale_mode has to be either 'min_max' or 'clip'.")
    return img
```

**Comment by jscanvic** [comment]
I've been working on merging those but now I'd argue we shouldn't exactly merge them.

Here's my understanding of it and what I believe we should do. Let me know if I'm missing something!

**What do the functions do?**

The function `preprocess_img` takes real and complex images. It takes the absolute value of complex images making them real-valued and further processes them like real input images. For real input images, it normalizes them between zero and one by rescaling or clamping them. Moreover, it assumes that dual channel real-valued tensors are Cartesian representations of complex images and treat them accordingly.

The function `rescale_img` takes real images but not complex images and it normalizes them between zero and one by rescaling or clamping them.

**How are they used?**

The function `preprocess_img` is only unsed in plotting functions, namely in `prepare_images`, `plot` and `plot_inset`.

The function `rescale_img` is used in plotting functions, namely in `plot_ortho3D` directly and in `preprocess_img` indirectly, but also in the dataset transform `Rescale` and in the `Metric` base class to normalize images before comparing them.

**Why not merging them?**

They serve two different purposes: `preprocess_img` pre-processes images for display but `rescale_img` is a broader image processing normalization operation.

**What to do instead?**

I suggest to:
* Move `rescale_img` out of the plotting module given it's used beyond plotting
* Make sure all plotting functions use `preprocess_img` since it's its purpose
* Let `preprocess_img` call `rescale_img` as it makes sense to normalize images before they get plotted
